### ISSUE TITLE
`pyenv install ...` not putting files in `~/.pyenv/bin`

### ISSUE THREAD

**Issue by MarcelRobitaille** [pyenv/pyenv#1753]
Too many issues will kill our team's development velocity, drastically.
Make sure you have checked all steps below.

### Prerequisite
* [x] Make sure your problem is not listed in [the common build problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems).
* [x] Make sure no duplicated issue has already been reported in [the pyenv issues](https://github.com/pyenv/pyenv/issues). You should look for closed issues, too.
* [x] Make sure you are not asking us to help solving your specific issue.
  * GitHub issues is opened mainly for development purposes. If you want to ask someone to help solving your problem, go to some community site like [Gitter](https://gitter.im/yyuu/pyenv), [StackOverflow](https://stackoverflow.com/questions/tagged/pyenv), etc.
* [x] Make sure your problem is not derived from packaging (e.g. [Homebrew](https://brew.sh)).
  * Please refer to the package documentation for the installation issues, etc.
* [x] Make sure your problem is not derived from plugins.
  * This repository is maintaining `pyenv` and the default `python-build` plugin only. Please refrain from reporting issues of other plugins here.

### Description
- [x] Platform information (e.g. Ubuntu Linux 16.04): Arch linux 5.9.12
- [x] OS architecture (e.g. amd64): amd64
- [x] pyenv version: 1.2.21
- [x] Python version: 3.9.0
- [x] C Compiler information (e.g. gcc 7.3): 10.2.0
- [x] Please attach verbose build log as gist: https://gist.github.com/MarcelRobitaille/1ec23767086621136fd2e4477ea60434
  * You can turn on verbose debug logging using by setting `PYENV_DEBUG=1`, e.g. `env PYENV_DEBUG=1 pyenv install -v 3.6.4`

`pyenv install ...` is putting files in `~/.pyenv/shims`, not `~/.pyenv/bin`. Adding the latter to `PATH` as the README suggests means an empty directory is added to `PATH` and `python` falls back to system python. The solution is to instead add `~/.pyenv/shims` to `PATH`.

**Comment by native-api** [comment]
`~/.pyenv/bin` is not empty, it contains (a symlink to) the main script, `pyenv`.

If it's empty for you, you didn't install Pyenv properly or something went wrong during installation.

**Comment by wyfo** [comment]
I've just installed pyenv with `brew install pyenv`, but I've no `~/.pyenv/bin` directory too. I had to add `~/.pyenv/shims` to path instead.
I didn't see errors in homebrew logs either.

**Comment by native-api** [comment]
> I've just installed pyenv with `brew install pyenv`, but I've no `~/.pyenv/bin` directory

This is normal. If you install with Homebrew, `pyenv` resides in Homebrew's `bin` directory.

**Comment by wyfo** [comment]
If it is normal, there should be an issue with the README. Quoting the installation instruction with Homebrew:
> Then follow the rest of the post-installation steps, starting with [Set up your shell environment for Pyenv](https://github.com/pyenv/pyenv?tab=readme-ov-file#set-up-your-shell-environment-for-pyenv).

In the following post-installation steps, you find various scripts like:
```bash
echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bashrc
echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bashrc
echo 'eval "$(pyenv init -)"' >> ~/.bashrc
```
As you see, `$PYENV_ROOT/bin` is mentioned in these scripts, and it fails with Homebrew installation.

Maybe we could add a comment into these scripts, like the following
```bash
echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bashrc
# If installed with Homebrew, `pyenv` resides in Homebrew's `bin` directory and not in `$PYENV_ROOT/bin`
echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bashrc
echo 'eval "$(pyenv init -)"' >> ~/.bashrc
```
I can open a PR if you want.

**Comment by native-api** [comment]
> ```shell
> echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bashrc
> ```

This is designed to work in both scenarios so that users don't have to think of which code to use. If Pyenv is installed with Homebrew, `pyenv`  is on `PATH` and nothing gets added.

**Comment by wyfo** [comment]
Indeed, sorry. I'm in fact using fish, and I didn't understand why `fish_add_path $PYENV_ROOT/bin` was failing, hence my initial comment.
Now, I understand that there is no issue, and your first answer will dismiss the misunderstanding of people like me. Sorry again.
